plugins/connection/incus.py

```diff
@@ -104,11 +104,11 @@ def __init__(self, play_context, new_stdin, *args, **kwargs):
         if getattr(self._shell, "_IS_WINDOWS", False):
             # Initializing regular expression patterns to match on a PowerShell or cmd command line.
             self.powershell_regex_pattern = re.compile(
-                r"^(?P<executable>(\"?([a-z]:)?[a-z0-9 ()\\.]+)?powershell(\.exe)?\"?|(([a-z]:)?[a-z0-9()\\.]+)?powershell(\.exe)?)\s+.*(?P<command>-c(ommand)?)\s+",  # noqa: E501
+                r'^"?(?P<executable>(?:[a-z]:\\)?[a-z0-9 ()\\.]*powershell(?:\.exe)?)"?\s+.*(?P<command>-c(?:ommand)?)\s+',
                 re.IGNORECASE,
             )
             self.cmd_regex_pattern = re.compile(
-                r"^(?P<executable>(\"?([a-z]:)?[a-z0-9 ()\\.]+)?cmd(\.exe)?\"?|(([a-z]:)?[a-z0-9()\\.]+)?cmd(\.exe)?)\s+.*(?P<command>/c)\s+",
+                r'^"?(?P<executable>(?:[a-z]:\\)?[a-z0-9 ()\\.]*cmd(?:\.exe)?)"?\s+.*(?P<command>/c)\s+',
```

Comment:
Also, as mentioned before, this part is not being captured:
```suggestion
                r'^"?(?P<executable>(?:[a-z]:\\)?[a-z0-9 ()\\.]*cmd(?:\.exe)?)"?\s+(?P<ignored>.*)(?P<command>/c)\s+',
```
and `_build_command()` obviously never uses it. Is that intentional? Don't we lose some parameters that might be important there?

Reply:
Sorry, I though you said that you understood and to forget your comment about this.

If you look at [these lines](https://github.com/ansible-collections/community.general/pull/11347/files#diff-7f31c09aee977074f8ac127e9b35b526768355ce1acae35822288db71bff0a93R155-R166) it builds the exec command by adding 

1. the executable first ([line 160](https://github.com/ansible-collections/community.general/pull/11347/files#diff-7f31c09aee977074f8ac127e9b35b526768355ce1acae35822288db71bff0a93R160))
2. then whatever is between the executable and `-c(ommand)?` (splitting on space) ([line 162](https://github.com/ansible-collections/community.general/pull/11347/files#diff-7f31c09aee977074f8ac127e9b35b526768355ce1acae35822288db71bff0a93R162))
3. then the command `-c(ommand)?` (PowerShell) or `/c` (cmd) ([line 164](https://github.com/ansible-collections/community.general/pull/11347/files#diff-7f31c09aee977074f8ac127e9b35b526768355ce1acae35822288db71bff0a93R164))
4. then anything remaining for the command as one block (no splitting on space) ([line 166](https://github.com/ansible-collections/community.general/pull/11347/files#diff-7f31c09aee977074f8ac127e9b35b526768355ce1acae35822288db71bff0a93R166))

So it is used later. At first, I did not want to put it in the regex to avoid any kind of problem mismatching or else. So I preferred to use Python to sort it.

Reply:
Sorry I wasn't clear - I was referring to something else then. Anyways, now I do see what you mean in the code, but that begs the question: since we are already running a regexp over that line, wouldn't it be much simpler to just capture that block instead of splitting (line 155), so that we slice and dice it in line 162?

Reply:
What I mean (pending verification of course) is something like:
https://github.com/ansible-collections/community.general/pull/11356/files